### DISCUSSION TITLE
How to use relative links instead of base_url in elevations.py?

### DISCUSSION THREAD

**Discussion by 4fd81048-Brian** [netbox-community/netbox#6674] [Q&A]
In my rack elevations, all the populated slots have a base_url set of the server's internal hostname (instead of the published name).  The unpopulated slots have the "add device" link and it is relative.

I don't have a base_url configured in configuration.py although I do have a BASE_PATH = 'netbox/'

Is there some way to make the populated slots in the svg relative links?

**Comment by 4fd81048-Brian** [comment]
I just commented out the block about self.base_url and set it to '' and it seems to have fixed it.

```
        #if base_url is not None:
        #    self.base_url = base_url.rstrip('/')
        #else:
        #    self.base_url = ''
        self.base_url = ''
```

**Comment by candlerb** [comment]
If you are getting the wrong hostname in the URLs generated for SVGs, it's because your proxy is setting the Host: header to be the proxy target, not the hostname from the original request.

The [sample configurations](https://github.com/netbox-community/netbox/tree/develop/contrib) for Apache and Nginx show how to do this properly.  For example, in Apache it's `ProxyPreserveHost On`

It *might* also work if your proxy sets the `X-Forwarded-Host` header.  From memory, I believe that Django will honour this, but not if it's proxied via two or more hops (i.e. if the header contains a list of hostnames instead of a single hostname)

**Reply by 4fd81048-Brian** [reply]
Thanks, I've now uncovered a lot of discussion on this issue.  Netbox is behind two proxies (one is Azure AD App Proxy which doesn't support the X-Forwarded-Host Header and the other is Apache on the box) so I'm looking for a work-around.  Isn't it odd that the unpopulated positions use relative links (fo add device) while the populated use fqdn?

**Reply by candlerb** [reply]
It would be nice if all the generated links were relative, but I don't know the underlying code reason why they're not.

~~A possible workaround on Apache might be (untested):~~

```
RequestHeader setifempty X-Forwarded-Host "netbox.example.com"
```

EDIT: I have tested this, and `X-Forwarded-Host` causes Netbox/Django to return a 400 error.  However this works:

```
RequestHeader set Host netbox.example.com
```

(as long as `netbox.example.com` is listed in `ALLOWED_HOSTS` in `configuration.py`)

**Reply by candlerb** [reply]
Also: I wonder if the reason the SVGs have absolute links is so that you can copy-paste them to some other server and the links to individual devices will still work.  If the links were relative, they would break if you put these images on your wiki (say).

As you've found, the absolute base_url is being [passed explicitly](https://github.com/netbox-community/netbox/blob/develop/netbox/dcim/api/views.py#L216) when drawing the elevations, so it seems to be intentional.

It does seem inconsistent that the "add device" links are still relative. Maybe you *don't* want those to work when the image is hosted remotely?
